connectors/citrus-openapi/src/main/java/org/citrusframework/openapi/OpenApiResourceLoader.java

```diff
@@ -52,68 +61,108 @@ private OpenApiResourceLoader() {
 
     /**
      * Loads the specification from a file resource. Either classpath or file system resource path is supported.
-     * @param resource
-     * @return
      */
     public static OasDocument fromFile(String resource) {
-        return fromFile(FileUtils.getFileResource(resource));
+        return fromFile(FileUtils.getFileResource(resource), OAS_RESOLVER);
     }
 
     /**
-     * Loads the specification from a file resource. Either classpath or file system resource path is supported.
-     * @param resource
-     * @return
+     * Loads the raw specification from a file resource. Either classpath or file system resource path is supported.
+     */
+    public static String rawFromFile(String resource) {
+        return fromFile(FileUtils.getFileResource(resource),
+            RAW_RESOLVER);
+    }
+
+    /**
+     * Loads the specification from a resource.
      */
     public static OasDocument fromFile(Resource resource) {
+        return fromFile(resource, OAS_RESOLVER);
+    }
+
+    /**
+     * Loads the raw specification from a resource.
+     */
+    public static String rawFromFile(Resource resource) {
+        return fromFile(resource, RAW_RESOLVER);
+    }
+
+    private static <T> T fromFile(Resource resource, Resolver<T> resolver) {
         try {
-            return resolve(FileUtils.readToString(resource));
+            return resolve(FileUtils.readToString(resource), resolver);
         } catch (IOException e) {
             throw new IllegalStateException("Failed to parse Open API specification: " + resource, e);
         }
     }
 
     /**
      * Loads specification from given web URL location.
-     * @param url
-     * @return
      */
     public static OasDocument fromWebResource(URL url) {
-        HttpURLConnection con = null;
+        return fromWebResource(url, OAS_RESOLVER);
+    }
+
+    /**
+     * Loads raw specification from given web URL location.
+     */
+    public static String rawFromWebResource(URL url) {
+        return fromWebResource(url, RAW_RESOLVER);
+    }
+
+    private static <T> T fromWebResource(URL url, Resolver<T> resolver) {
+        URLConnection con = null;
         try {
-            con = (HttpURLConnection) url.openConnection();
-            con.setRequestMethod(HttpMethod.GET.name());
-            con.setRequestProperty(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+            con = url.openConnection();
+
+            if (con instanceof HttpURLConnection httpURLConnection) {
+                httpURLConnection.setRequestMethod(HttpMethod.GET.name());
+                con.setRequestProperty(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+
+                int status = httpURLConnection.getResponseCode();
+                if (status > 299) {
```

Comment:
I disagree:
Given the default behavior of HttpURLConnection, it automatically follows redirects, so the status code we receive will be from the final response, typically below 300. If we receive a 3xx status that is not automatically followed, it must be handled explicitly here. Since we are not handling such cases, receiving an unhandled 3xx status results in an unprocessed response, justifying the thrown exception.

Reply:
that is true, it should follow automatically. didn't consider that.